### ISSUE TITLE
Importers / Processors not showing

### ISSUE THREAD

**Issue by joshk418** [craftworkgames/MonoGame.Extended#626]
Sorry if this has been posted elsewhere, but I was having issues with the Importers and processors not showing up in the content pipeline. I figured this might be due to some recent changes.

I'm using Visual Studio Community 2017 with .NET 4.8, Monogame 3.7.1, and Monogame.Exteneded 3.7.0

Here are my references in the pipeline:
 * /reference:..\..\packages\MonoGame.Extended.Content.Pipeline.3.7.0\lib\netstandard2.0\MonoGame.Extended.Content.Pipeline.dll
 * /reference:..\..\packages\MonoGame.Extended.Tiled.3.7.0\lib\netstandard2.0\MonoGame.Extended.Tiled.dll

![references](https://user-images.githubusercontent.com/22644674/63899401-bfcd7280-c9ca-11e9-9022-19ce060d1ef4.PNG)

Here's my importer selection:

![importers](https://user-images.githubusercontent.com/22644674/63899395-b93efb00-c9ca-11e9-8372-27d03a7440ac.PNG)

**Comment by craftworkgames** [comment]
This is probably the same issue that came up on the [community forums recently](http://community.monogame.net/t/solved-couldnt-find-a-default-importer-for-tmx-files/11782/5).

I have implemented part of the fix in the `develop` branch by packaging the `MonoGame.Extended.Content.Pipeline` package differently. It's quite a complex issue though, so I'm not sure if that's enough to resolve it.

Could I get you to try building from source (just clone the repo and run `build.ps1`) and see if that resolves the issue for you?

**Comment by joshk418** [comment]
That seemed to have worked well by building from the master branch. The build.ps1 kept failing for some reason, but I just built each project in the solution individually and stored the DLLs in my game's root directory. Thanks for the help!

Also let me know if it was a bad idea to not use the build.ps1 (not sure if I missed important steps the way I did it)

**Comment by craftworkgames** [comment]
> The build.ps1 kept failing for some reason

Depends what the reason is I guess? If you post the error I should be able to explain what's going on.

I just ran it on a different computer and it worked okay. So in theory it should work for you. Although, there are some dependencies you need, like Visual Studio and you must to run it from a git repository.

> I just built each project in the solution individually and stored the DLLs in my game's root directory.

There are many ways to get it to work and if you're happy with it then it's fine. The main problem with this approach is manually copying DLL's around can bit a pretty annoying if you need to do it a lot.

> Also let me know if it was a bad idea to not use the build.ps1 (not sure if I missed important steps the way I did it)

The cake script (which is what runs when you execute `build.ps1`) does the following things:
 - generates a version number (from the commits in the git repository)
 - builds all the projects (Visual Studio will do this too)
 - runs the unit tests
 - creates NuGet packages

So basically, the main benefit is that you can create your own properly versioned NuGet packages locally and reference them that way. 

The actual DLL's generated will be pretty much the same in every other way though..

**Comment by joshk418** [comment]
Ahh that was the problem. For some reason, I downloaded the repo instead of cloning it. It works after cloning it like you said. My bad haha. Thanks again for the help

**Comment by vividos** [comment]
I had the same problem that @joshk418 had; it seems that when the content build tool wants to load the reference, it doesn't find the referenced assemblies (mostly the MonoGame.Framework assemblies). I could solve the issue by putting the correct versions of the following assemblies into the `monogame.extended.content.pipeline\3.7.0\lib\netstandard2.0` folder: MonoGame.Extended.dll, MonoGame.Extended.Tiled.dll, MonoGame.Framework.Content.Pipeline.dll and MonoGame.Framework.dll. I was just compiling a Tiled map, so for other types there may be more .dlls from MonoGame.Extended needed.

**Comment by fossegutten** [comment]
I also encountered this yesterday. The way i fixed/worked around it was:
1. Install nuget package for Monogame extended content pipeline.
2. Add all required dll files in one folder, inside the project: (tiled, monogame extended content pipeline, monogame extended, monogame extended graphics). Im not sure all these are needed, i will test further later.
3. Add monogame extended content pipeline dll as a reference, in Content pipeline program.
4. Importer for tiled worked, without pipeline crashing.

It seems to be a problem with monogame content pipeline, for all dll references that depend on other dll. There is an issue over at their repo too. I´m no expert in .net or C#, but one way to workaround could be to distribute the dll in one folder, inside the nuget package?

**Comment by chrisevans9629** [comment]
I think I am having the same issue as well.  I made an issue on the monogame project [here](https://github.com/MonoGame/MonoGame/issues/7013).  I'll try @fossegutten work around when I get a chance to

**Comment by QuinnBast** [comment]
This is duplicated in #6942 as well. Surprised this hasn't been fixed yet as it prevents any monogame pipline references being loaded.

**Comment by fossegutten** [comment]
@QuinnBast I saw your post at monogame repo. You can see my workaround in the post above. It only takes a few minutes, hope it works for you guys also 😄

**Comment by klnsem** [comment]
If you've already added the .dll-reference the way it's spelled out [here](http://docs.monogameextended.net/Installation/#referencing-the-content-pipeline-extension), the Pipeline Tool can't remove it. _This is a issue with MonoGame, not .Extended_. To fix this: 
-- First, open Content.mgcb in Notepad, and remove this part, between the "References" and "Content"-lines:
```
#-------------------------------- References --------------------------------#

Remove Everything here.

#---------------------------------- Content ---------------------------------#
```
-- Save the file in Notepad. 

-- Then do what fossegutten [suggests](https://github.com/craftworkgames/MonoGame.Extended/issues/626#issuecomment-583733555). 

**Version info:**
```
  <package id="MonoGame.Extended" version="3.7.0" targetFramework="net472" />
  <package id="MonoGame.Extended.Content.Pipeline" version="3.7.0" targetFramework="net472" />
  <package id="MonoGame.Extended.Graphics" version="3.7.0" targetFramework="net472" />
  <package id="MonoGame.Extended.Tiled" version="3.7.0" targetFramework="net472" />
  <package id="MonoGame.Framework.Content.Pipeline.Portable" version="3.7.1.189" targetFramework="net472" />
  <package id="MonoGame.Framework.Portable" version="3.7.1.189" targetFramework="net472" />
```

(I haven't posted on an issue before, so if I've committed any faux pas, it's not with intent)

**Comment by sudo-make** [comment]
Was trying to follow @fossegutten 's steps; but I cannot for the life of me find where this "Reference Editor" tool is.

**Comment by Gixen** [comment]
> Was trying to follow @fossegutten 's steps; but I cannot for the life of me find where this "Reference Editor" tool is.

I think that's the one in the pipeline tool

**Comment by lithiumtoast** [comment]
I would be OK with just adding/updating the documentation about "How to use content pipeline with MonoGame.Extended" to resolve this issue

**Comment by QuimbleyStudio** [comment]
Experiencing this issue as well and still after following a few work arounds,  including fossegottens, I am still unable to build the TMX files. At a a loss on what to do here other then just not use Tiled.

**Comment by jamesdolan16** [comment]
Any news guys, experiencing the exact same issue :/

**Comment by lithiumtoast** [comment]
As it was mentioned before, this issue is not a MonoGame.Extended issue and more of a MonoGame issue.

I have set up a repo to which the problem is resolved: https://github.com/craftworkgames/MonoGame.Extended-samples

The problem is that MCGB does not understand NuGet packages. To overcome this the `.mgcb` needs a path reference to `.dll` from the NuGet package. In the above repo, I altered the path for NuGet packages to get restored and edited the `.mgcb` files to point to this path. It's possible to not alter the NuGet packages restore path ad use the default path in yor `.mgcb` files.

**Comment by janfokke** [comment]
Since there is quite a bit of confusion about adding references, I made a [video ](https://www.youtube.com/watch?v=uJ0chFOWx3k) that shows how to do this

Note. I am using MonoGame 3.8
[Here](https://docs.monogame.net/articles/getting_started/1_setting_up_your_development_environment_windows.html) is the explanation on how to install it

**Comment by bowbriers** [comment]
I've read through this thread and attempted all steps here, including the steps from fossegutten and going through the video from janfokke.  Unfortunately, I still cannot import a fnt file.  The Build Output indicates, "Couldn't find a default importer for '[path]/OpenSans32.fnt'!  Also, the Settings for the .fnt file indicate "Invalid / Missing Importer" and "Invalid / Missing Processor".  Please note that all five referenced dll files are in the same folder (Content/References) in my Project directory.  Any suggestions?
Note that I also cannot manually select the BMFontImporter or BMFontProcessor options that seem to be indicated by the instructions here: [https://www.monogameextended.net/docs/features/bitmap-font/bitmap-font/](https://www.monogameextended.net/docs/features/bitmap-font/bitmap-font/).

Apologies if I'm missing something obvious here - I'm relatively new to Monogame and quite new to Monogame Extended.
 
![image](https://user-images.githubusercontent.com/38353135/100926850-0a814880-34b2-11eb-854e-947e3367f133.png)
![image](https://user-images.githubusercontent.com/38353135/100926881-14a34700-34b2-11eb-8620-7f8af4b0ae60.png)
![image](https://user-images.githubusercontent.com/38353135/100927134-74015700-34b2-11eb-9b5e-2dd2756ebfcf.png)

**Comment by bowbriers** [comment]
Update to my previous issue - my issue seems to be resolved by upgrading from Visual Studio 2017 to Visual Studio 2019 (presumably because VS 2017 did not support .NET Core 3.1, which is a prerequisite for version 3.8.0 of MonoGame.Extended)

**Comment by slanger** [comment]
@lithiumtoast's work-around in [this comment](https://github.com/craftworkgames/MonoGame.Extended/issues/626#issuecomment-720449736) worked for me; see https://github.com/slanger/zelda-online/commit/f060f71994ba266163fd71277dc3fe03dacc9311. I'm happy with the result.

Looks like the documentation was updated with this work-around in https://github.com/craftworkgames/craftworkgames.github.io/commit/972ceaa330cfb1428d77c282331a8314de83dc1f. (Though, it's probably worth adding that a `nuget.config` file can move the location of NuGet packages to a more convenient location.)

**Comment by AristurtleDev** [comment]
This issue no longer exists in the current `develop` branch.
Please either use `develop` or the most recent prerelase NuGet
